src/main/java/com/epam/test_generator/entities/api/CaseTrait.java

```diff
@@ -1,18 +1,45 @@
 package com.epam.test_generator.entities.api;
 
+import com.epam.test_generator.entities.Status;
 import com.epam.test_generator.entities.Step;
+import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
 
 public interface CaseTrait {
 
     List<Step> getSteps();
 
+    void setSteps(List<Step> objects);
+
+    void setStatus(Status status);
+
     default boolean hasStep(Step step) {
         return getSteps().contains(step);
     }
 
     default boolean removeStep(Step step) {
-        return getSteps().remove(step);
+        boolean isRemoved = getSteps().remove(step);
+        if(getSteps().isEmpty()){
```

Comment:
According to Daria, when there is more then 1 steps left, status should be set to NOT_RUN.